File: src/logic/logEntries.ts
import { LogEventKey, LogParamKey, type LogEntry, type Player } from '../types/game';

export function actorEntry(
  key: LogEventKey,
  player: Player,
  extra: Record<string, string | number> = {},
): LogEntry {
  return {
    key,
    params: { name: player.name, ...(player.botControlled ? { [LogParamKey.Bot]: true } : {}), ...extra },
  };
}

export function turnEntry(players: Player[], nextId: number): LogEntry {
  const p = players[nextId];
  return { key: LogEventKey.Turn, params: { name: p.name, ...(p.botControlled ? { [LogParamKey.Bot]: true } : {}) } };
}

File: backend/internal/usecase/gameusecase/log_entries.go
package gameusecase

import "monopoly-game-backend/internal/entity"

func ActorEntry(key entity.LogEventKey, player entity.Player, extra map[string]interface{}) entity.LogEntry {
	params := map[string]interface{}{
		"name": player.Name,
	}
	if player.BotControlled {
		params[entity.LogParamKeyBot] = true
	}
	for k, v := range extra {
		params[k] = v
	}
	return entity.LogEntry{Key: key, Params: params}
}

func TurnEntry(players []entity.Player, nextID int) entity.LogEntry {
	p := players[nextID]
	params := map[string]interface{}{
		"name": p.Name,
	}
	if p.BotControlled {
		params[entity.LogParamKeyBot] = true
	}
	return entity.LogEntry{Key: entity.LogEventKeyTurn, Params: params}
}
